### PR TITLE
Add identify

### PR DESCRIPTION
**Pull request by acured**
Please use `[API Name] Brief description` as title.

## API Name (Kind)
`Face`, `TextAnalytics`, `ComputerVision`...

## Purpose
Describe the intention of the changes being proposed.
What problem does it solve or functionality does it add?

## Other Information
Add any other helpful information that may be needed here.

### CODE REVIEW

File: Face/Identify/Program.cs [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+
+            // Delete the person group.
+            await Client.PersonGroup.DeleteAsync(GroupId);
+            Console.WriteLine($"{Environment.NewLine}Delete the person group {GroupId}{Environment.NewLine}"); 
+        }
+
+        public static async Task Identify_in_LargePersonGroup()
+        {
+            // Create a large person group.
+            string GroupId = Guid.NewGuid().ToString();
```
Comment (lebronJ) [inline Face/Identify/Program.cs]: `largePersonGroupId`

File: Face/Identify/Program.cs [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+            Console.WriteLine($"Create a large person group ({GroupId}) {Environment.NewLine}");
+            await Client.LargePersonGroup.CreateAsync(GroupId, GroupId);
+
+            foreach (var targetImageFileName in TargetImageFileNames)
+            {
+                // Read image file. 
+                using (FileStream stream = new FileStream(Path.Combine("Images", targetImageFileName), FileMode.Open))
+                {
+                    // Create a person group person.
+                    Person p = new Person();
```
Comment (lebronJ) [inline Face/Identify/Program.cs]: Person p = new Person { Name = targetImageFileName, UserData = "Person for sample" };

File: Face/Identify/Program.cs [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+                Console.WriteLine($"Response: Success. Group \"{GroupId}' training process is {status.Status}");
+
+                if (status.Status != TrainingStatusType.Running)
+                {
+                    break;
+                }
+            }
+
+            List<Guid> faceIds = new List<Guid>();
+            // Read image file.
```
Comment (lebronJ) [inline Face/Identify/Program.cs]: Newline before comments.

File: Face/Identify/Program.cs [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+                if (detectedFaces[0].FaceId == null)
+                {
+                    Console.WriteLine("[Error] Parameter `returnFaceId` of `DetectWithStreamAsync` must be set to `true` (by default) for Identify purpose.");
+                    return;
+                }
+
+                // Add detected faceId to faceIds.
+                foreach (var face in detectedFaces)
+                {
+                    faceIds.Add(face.FaceId.Value);
```
Comment (lebronJ) [inline Face/Identify/Program.cs]: `FaceId` can be null.

foreach (var face in detectedFaces)
                {
                    if (face.FaceId != null)
                    {
                        faceIds.Add(face.FaceId.Value);
                    }
                }

File: Face/Identify/Program.cs [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+                {
+                    // Create a person group person.
+                    Person p = new Person();
+                    p.Name = targetImageFileName;
+                    p.UserData = "Person for sample";
+                    Console.WriteLine($"Create a person group person '{p.Name}'");
+                    p.PersonId = (await Client.PersonGroupPerson.CreateAsync(GroupId, p.Name)).PersonId;
+
+                    // Add face to the person group.
+                    Console.WriteLine($"Add faces to the person group from image `{targetImageFileName}`. {Environment.NewLine}");
```
Comment (lebronJ) [inline Face/Identify/Program.cs]: `faces` to `face` for clarification.
Please help modify all same issues.

File: Face/Identify/Program.cs [hunk 6]
```diff
@@ -0,0 +1,218 @@
...
+
+            // Start to train the person group.
+            await Client.PersonGroup.TrainAsync(GroupId);
+
+            // Wait until the training is completed.
+            while (true)
+            {
+                await Task.Delay(1000);
+                var status = await Client.PersonGroup.GetTrainingStatusAsync(GroupId);
+                Console.WriteLine($"Response: Success. Group \"{GroupId}' training process is {status.Status}");
```
Comment (lebronJ) [inline Face/Identify/Program.cs]: Please use the same output for getting training status as `FindSimilar` sample.

File: Face/Identify/Program.cs (hunk 5 above)
Comment (lebronJ) [inline Face/Identify/Program.cs]: `to the person group person`
Please help modify all same issues.

File: Face/Identify/Program.cs [hunk 7]
```diff
@@ -0,0 +1,218 @@
...
+            if (identifyResults == null)
+            {
+                Console.WriteLine($"No people found in the group the same as the {SourceImageFileName}.");
+                return;
+            }
+
+            foreach (var identifyResult in identifyResults)
+            {
+                Person person = await Client.PersonGroupPerson.GetAsync(GroupId, identifyResult.Candidates[0].PersonId);                
+                Console.WriteLine($"Person from {SourceImageFileName} & group person '{person.Name}' are of the same (Positive) person, confidence: {identifyResult.Candidates[0].Confidence}.");
```
Comment (lebronJ) [inline Face/Identify/Program.cs]: `Console.WriteLine($"Person '{person.Name}' is identified for faces in {SourceImageFileName}, confidence: {identifyResult.Candidates[0].Confidence}.");`

File: Face/Identify/Program.cs [hunk 8]
```diff
@@ -0,0 +1,218 @@
...
+            foreach (var targetImageFileName in TargetImageFileNames)
+            {
+                // Read image file. 
+                using (FileStream stream = new FileStream(Path.Combine("Images", targetImageFileName), FileMode.Open))
+                {
+                    // Create a person group person.
+                    Person p = new Person();
+                    p.Name = targetImageFileName;
+                    p.UserData = "Person for sample";
+                    Console.WriteLine($"Create a person group person '{p.Name}'");
```
Comment (lebronJ) [inline Face/Identify/Program.cs]: `large person group person`

File: Face/Identify/Program.cs [hunk 9]
```diff
@@ -0,0 +1,218 @@
...
+                    p.PersonId = (await Client.LargePersonGroupPerson.CreateAsync(GroupId, p.Name)).PersonId;
+
+                    // Add face to the large person group.
+                    Console.WriteLine($"Add faces to the large person group from image `{targetImageFileName}`.{Environment.NewLine}");
```
Comment (lebronJ) [inline Face/Identify/Program.cs]: `Add face to the large person group person from image `{targetImageFileName}.``

File: Face/Identify/Identify.csproj [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
+    </None>
+    <None Update="Images\Family2-Lady1.jpg">
+      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
+    </None>
+    <None Update="Images\Family2-Man1.jpg">
+      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
+    </None>
+    <None Update="Images\Family3-Lady1.jpg">
+      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
```
[the remark this answers is not in the document]
Comment (acured) [reply Face/Identify/Identify.csproj]: done

File: Face/Identify/Program.cs [hunk 11]
```diff
@@ -0,0 +1,218 @@
...
+using System.Threading.Tasks;
+using Microsoft.Azure.CognitiveServices.Vision.Face;
+using Microsoft.Azure.CognitiveServices.Vision.Face.Models;
+
+namespace Identify
+{
+    public class Program
+    {
+        // Create a client.
+        static string ApiKey = "ENTER YOUR KEY HERE";
```
[the remark this answers is not in the document]
Comment (acured) [reply Face/Identify/Program.cs]: done

File: Face/Identify/Program.cs [hunk 12]
```diff
@@ -0,0 +1,218 @@
...
+
+        static IFaceClient Client = new FaceClient(new ApiKeyServiceClientCredentials(ApiKey))
```
[the remark this answers is not in the document]
Comment (acured) [reply Face/Identify/Program.cs]: done

File: Face/Identify/Program.cs [hunk 13]
```diff
@@ -0,0 +1,218 @@
...
+            Identify_in_LargePersonGroup().Wait();
+
+            Console.WriteLine("\nPress ENTER to exit.");
+            Console.ReadLine();
+        }
+
+        public static async Task Identify_in_PersonGroup() 
+        {
+            // Create a person group.
+            string GroupId = Guid.NewGuid().ToString();
```
[the remark this answers is not in the document]
Comment (acured) [reply Face/Identify/Program.cs]: done

File: Face/Identify/Program.cs [hunk 14]
```diff
@@ -0,0 +1,218 @@
...
+            Console.WriteLine($"Create a person group ({GroupId}) {Environment.NewLine}");
```
[the remark this answers is not in the document]
Comment (acured) [reply Face/Identify/Program.cs]: done

File: Face/Identify/Program.cs [hunk 15]
```diff
@@ -0,0 +1,218 @@
...
+
+            foreach (var targetImageFileName in TargetImageFileNames) 
+            { 
+                // Read image file. 
+                using (FileStream stream = new FileStream(Path.Combine("Images", targetImageFileName), FileMode.Open)) 
+                {
+                    // Create a person group person.
+                    Person p = new Person();
+                    p.Name = targetImageFileName;
+                    p.UserData = "Person for sample";
```
[the remark this answers is not in the document]
Comment (acured) [reply Face/Identify/Program.cs]: done

File: Face/Identify/Program.cs [hunk 16]
```diff
@@ -0,0 +1,218 @@
...
+                Console.WriteLine($"Response: Success. Group \"{GroupId}' training process is {status.Status}");
+
+                if (status.Status != TrainingStatusType.Running)
+                {
+                    break;
+                }
+            }
+
+            List<Guid> faceIds = new List<Guid>();
+            // Read image file.
```
[the remark this answers is not in the document]
Comment (acured) [reply Face/Identify/Program.cs]: done

File: Face/Identify/Program.cs [hunk 17]
```diff
@@ -0,0 +1,218 @@
...
+                if (detectedFaces[0].FaceId == null) 
+                { 
+                    Console.WriteLine("[Error] Parameter `returnFaceId` of `DetectWithStreamAsync` must be set to `true` (by default) for Identify purpose.");
+                    return; 
+                }
+
+                // Add detected faceId to faceIds.
+                foreach (var face in detectedFaces)
+                {
+                    faceIds.Add(face.FaceId.Value);
```
[the remark this answers is not in the document]
Comment (acured) [reply Face/Identify/Program.cs]: done

File: Face/Identify/Program.cs (hunk 1 above)
Comment (acured) [reply Face/Identify/Program.cs]: done

File: Face/Identify/Program.cs (hunk 2 above)
Comment (acured) [reply Face/Identify/Program.cs]: done

File: Face/Identify/Program.cs (hunk 3 above)
Comment (acured) [reply Face/Identify/Program.cs]: done

File: Face/Identify/Program.cs (hunk 4 above)
Comment (acured) [reply Face/Identify/Program.cs]: done

File: Face/Identify/Program.cs (hunk 5 above)
Comment (acured) [reply Face/Identify/Program.cs]: done

File: Face/Identify/Program.cs (hunk 6 above)
Comment (acured) [reply Face/Identify/Program.cs]: done

File: Face/Identify/Program.cs (hunk 5 above)
Comment (acured) [reply Face/Identify/Program.cs]: done

File: Face/Identify/Program.cs (hunk 7 above)
Comment (acured) [reply Face/Identify/Program.cs]: done

File: Face/Identify/Program.cs (hunk 8 above)
Comment (acured) [reply Face/Identify/Program.cs]: done

File: Face/Identify/Program.cs (hunk 9 above)
Comment (acured) [reply Face/Identify/Program.cs]: done

**Pull request merged**